File: frontend/src/app/models/carInfo.ts
import {CarInOrder} from "./CarInOrder";

export class CarInfo {
    carId: string;
    carName: string;
    milesperGallon: number;
    cylinders: number;
    displacement: number;
    carImage: string;
    horsepower: number;
    carPrice: number;
    acceleration: number;
    licensed: boolean;
    addedDate: Date;
    warehouse: string


    constructor(carInOrder?: CarInOrder) {
            this.carId = carInOrder.carId;
            this.carName = carInOrder.carName;
            this.milesperGallon = carInOrder.milesperGallon;
            this.cylinders = carInOrder.cylinders;
            this.displacement = carInOrder.displacement;
            this.carImage = carInOrder.carImage;
            this.horsepower = carInOrder.horsepower;
            this.carPrice = carInOrder.carPrice;
            this.acceleration = carInOrder.acceleration;
            this.licensed = carInOrder.licensed;
            this.addedDate = carInOrder.addedDate;
            this.warehouse = carInOrder.warehouse;
    }
}

File: backend/src/main/java/com/cognizant/garage/entity/CarInfo.java
package com.cognizant.garage.entity;

import lombok.Data;
import org.hibernate.annotations.DynamicUpdate;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

@Entity
@Data
@DynamicUpdate
@Table(name = "car_info")
public class CarInfo implements Serializable {
    @Id
    private String carId;

    @NotNull
    private String carName;

    private Double milesperGallon;

    private Integer cylinders;
    private Integer displacement;
    private String carImage;
    private Integer horsepower;
    private BigDecimal carPrice;
    private Double acceleration;
    private Boolean licensed;

    private Date addedDate;

    private String warehouse;
    public CarInfo() {
    }
}
